### PR TITLE
Servers can't delay if Early-Data is present

### PR DESCRIPTION
**Pull request by martinthomson**
Moved from #27.

### CODE REVIEW

**Comment by wtarreau** [comment]
I think it would be simpler and possibly more accurate to say something along : 

A request containing an Early-Data header field cannot be turned into a safe request anymore even if the server waits for a possible ClientFinished signal. This request has already transited as unverified early data on a previous hop. The server may only use 4NN to request that it is retried without using early data.

**Comment by martinthomson** [comment]
Take 2:

> A server cannot make a request that contains the Early-Data header field safe for processing by waiting for the handshake to complete.  A request that is marked with Early-Data was sent in early data on a previous hop.  Requests that contain the Early-Data field and cannot be safely processed MUST be rejected using the 4NN (Too Early) status code.

**Comment by wtarreau** [comment]
On Fri, Aug 04, 2017 at 07:42:13AM +0000, Martin Thomson wrote:
> Take 2:
> 
> > A server cannot make a request that contains the Early-Data header field safe for processing by waiting for the handshake to complete.  A request that is marked with Early-Data was sent in early data on a previous hop.  Requests that contain the Early-Data field and cannot be safely processed MUST be rejected using the 4NN (Too Early) status code.

Perfect!

Willy

File: draft-thomson-http-replay.md [hunk 1]
```diff
@@ -257,6 +257,14 @@ that the client understands this specification and is willing to retry a request
 in response to a 4NN (Too Early) status code.  A user agent that sends a request
 in early data does not need to include the `Early-Data` header field.
 
+A server that receives a request marked with the Early-Data header field cannot
+wait until the TLS handshake completes in order to render the request safe from
+risk of replay.  The Early-Data header field indicates that early data was used
```
Comment (kaduk) [inline draft-thomson-http-replay.md]: "risk of replay" is ambiguous about "risk of this request being a replay" vs. "risk that this request could be replayed elsewhere/elsewhen".  The latter is nonsensical to an informed reader, of course, but we may not want to rely on that.  Take 2 in the comments is better in this regard, but is perhaps at risk of causing "safe" to have too many overloaded meanings.

Comment (martinthomson) [reply draft-thomson-http-replay.md]: I used a qualified safe.  I think that's safe.

**Pull request merged**